Support concurrency

dav3r opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adds concurrent fetching of CVE URLs and concurrent processing of the JSON from those URLs to load the CVE data into the CyHy database.

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Without parallelization, it was taking over 20 minutes to do an initial load of every CVE (roughly 236,000 as of today) into a new CyHy database.  We would like to run this process from an AWS Lambda function (15 minute maximum run time), so the goal of this PR is to speed things up.  Testing showed that the changes in this PR brought the runtime down to about 11 minutes when [running this code in a Lambda](https://github.com/cisagov/cyhy-cvesync-lambda) against a fresh database that had no CVEs in it.

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

I tested these changes by running all of the unit tests, including the one that loads every CVE from the NVD (~236k)  into the database, and everything works as expected.

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [x] Create `v1.1.0` release.
- [x] Move `v1` tag (`git pull; git tag -d v1; git tag v1 v1.1.0; git push --force --tags`)

dav3r committed:
Add aiohttp to requirements

This library handles asynchronous HTTP requests.

dav3r committed:
Add url_concurrency to the config model and a corresponding unit test

url_concurrency controls the number of concurrent URL requests to fetch and process CVE data.

dav3r committed:
Update cve_sync functions to asynchronously and concurrently fetch CVE URLs and process their JSON data

dav3r committed:
Update unit tests to support concurrent/async changes

dav3r committed:
Bump version from 1.0.0 to 1.1.0

dav3r committed:
Update example code and output in README to reflect latest changes

jsf9k reviewed (approved):
I also verified that I could run `pytest -vs --mongo-express` locally and all tests passed.

felddy reviewed (approved):
Great improvement!  🌟 
The new threading brought this scene to mind: 🏒 

![miracle-herb-brooks](https://github.com/user-attachments/assets/41abadd1-3222-4448-ac37-d54cbb92ab03)

dav3r committed:
Remove an unnecessary "Optional" type hint

This was causing mypy 1.13 to throw an error like this:

src/cyhy_cvesync/main.py:63: error: Argument 3 to "process_urls" has incompatible type "int | None"; expected "int"  [arg-type]

The current version of mypy used in by pre-commit is 1.10.0 and it was NOT flagging this error, but my local mypy 1.13.0 was so I went ahead and fixed it.  All pytests still pass after this change.

jsf9k commented:
Please make the "Run diagnostics" checks required.
